### PR TITLE
New rule: use_rethrow_when_possible

### PR DESCRIPTION
**Pull request by JPaulsen**
Fixes https://github.com/dart-lang/linter/issues/518

### CODE REVIEW

File: test/rules/use_rethrow_when_possible.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  }
+}
+
+void bad2() {
+  try {} catch (e, stackTrace) {
+    print(stackTrace);
+    throw e; // LINT
+  }
+}
+
```
Comment (bwilkerson) [inline test/rules/use_rethrow_when_possible.dart]: Please add this test case:
```
void bad3() {
  try {} catch (e) {
    try {} catch (f) {
      throw e; // LINT
    }
  }
}
```
unless you think this shouldn't produce a lint.

Comment (JPaulsen) [inline test/rules/use_rethrow_when_possible.dart]: I think it should not produce a lint, but I can add this test:

```dart
void good3() {
  try {} catch (e) {
    try {} catch (f) {
      throw e; // OK
    }
  }
}
```

Comment (bwilkerson) [inline test/rules/use_rethrow_when_possible.dart]: Ah, you're right. I forgot that rethrow implicitly applies to the exception caught by the most closely surrounding catch clause.

Comment (JPaulsen) [inline test/rules/use_rethrow_when_possible.dart]: Ok, thanks for the review, let me add the test.

Comment (JPaulsen) [inline test/rules/use_rethrow_when_possible.dart]: Fixed (test good3 added)

**Pull request merged**